resteasy-spring-boot-starter/src/main/java/org/jboss/resteasy/springboot/ResteasyAutoConfiguration.java

```diff
@@ -39,7 +40,7 @@
     @Bean
     @Qualifier("ResteasyProviderFactory")
     public static BeanFactoryPostProcessor springBeanProcessor() {
-        ResteasyProviderFactory resteasyProviderFactory = new ResteasyProviderFactory();
+        ResteasyProviderFactory resteasyProviderFactory = new ResteasyProviderFactoryImpl();
```

Comment:
@liweinan you should actually do ``ResteasyProviderFactory.newInstance()`` here.

Reply:
okay!

Reply:
okay!